src/Tvheadend.cpp

```diff
@@ -1586,12 +1586,27 @@ void CTvheadend::OnWake()
 
 bool CTvheadend::VfsOpen(const PVR_RECORDING &rec)
 {
-  return m_vfs->Open(rec);
+  bool ret = m_vfs->Open(rec);
+
+  if (ret)
+  {
+    const auto &it = m_recordings.find(atoi(rec.strRecordingId));
+    if (it != m_recordings.end())
+    {
+      CLockObject lock(m_mutex);
+      m_playingRecording = &(it->second);
```

Comment:
Not sure if it's possible but what if the currently playing recording is deleted? Won't m_playingRecording point to a stale pointer and GetStreamTimes will crash?

Reply:
It would, but imo the recording that is currently played cannot be deleted.

Reply:
In my first version I had a member storing the recording id and every GetStreamTimes included a map lookup to find the recording. I thought this would be to expensive.

Reply:
I guess this is fine then

Reply:
Anyway, I will check again and maybe try to make it more failsafe, just to be sure.